### ISSUE TITLE
[Linux/arm64] NativeLibraryTests_sh failure

### ISSUE THREAD

**Issue by BruceForstall** [dotnet/coreclr#22365]
COMPlus_GCStress=0xC
COMPlus_TieredCompilation=0

https://ci.dot.net/job/dotnet_coreclr/job/master/job/jitstress/job/arm64_cross_checked_ubuntu16.04_gcstress0xc_tst/17/consoleFull

I can repro this locally about every 4 runs.

```
Interop_NativeLibrary._NativeLibraryTests_NativeLibraryTests_._NativeLibraryTests_NativeLibraryTests_sh [FAIL]
01:45:43         /n/nReturn code:      1/nRaw output file:      /home/robox/j/workspace/dotnet_coreclr/master/jitstress/arm64_cross_checked_ubuntu16.04_gcstress0xc_tst/bin/tests/Linux.arm64.Checked/Interop/NativeLibrary/Reports/Interop.NativeLibrary/NativeLibraryTests/NativeLibraryTests.output.txt/nRaw output:/nBEGIN EXECUTION
01:45:43         /home/robox/j/workspace/dotnet_coreclr/master/jitstress/arm64_cross_checked_ubuntu16.04_gcstress0xc_tst/bin/tests/Linux.arm64.Checked/Tests/Core_Root/corerun NativeLibraryTests.exe
01:45:43          FreeLibrary(188350176) : Success : [FAIL]
01:45:43         Expected: 100
01:45:43         Actual: 156
01:45:43         END EXECUTION - FAILED
01:45:43         Test Harness Exitcode is : 1
01:45:43         /nTo run the test:/n> set CORE_ROOT=/home/robox/j/workspace/dotnet_coreclr/master/jitstress/arm64_cross_checked_ubuntu16.04_gcstress0xc_tst/bin/tests/Linux.arm64.Checked/Tests/Core_Root/n> /home/robox/j/workspace/dotnet_coreclr/master/jitstress/arm64_cross_checked_ubuntu16.04_gcstress0xc_tst/bin/tests/Linux.arm64.Checked/Interop/NativeLibrary/NativeLibraryTests/NativeLibraryTests.sh/n
01:45:43         Expected: True
01:45:43         Actual:   False
01:45:43         Stack Trace:
01:45:43           /mnt/j/workspace/dotnet_coreclr/master/arm64_cross_checked_ubuntu16.04/bin/tests/Linux.arm64.Checked/TestWrappers/Interop.NativeLibrary/Interop.NativeLibrary.XUnitWrapper.cs(109,0): at Interop_NativeLibrary._NativeLibraryTests_NativeLibraryTests_._NativeLibraryTests_NativeLibraryTests_sh()
```

**Comment by BruceForstall** [comment]
@AaronRobinsonMSFT @jkoritzinsky

**Comment by AaronRobinsonMSFT** [comment]
@swaroop-sridhar Can you take a first stab at this issue?

**Comment by swaroop-sridhar** [comment]
Sure, I'll take a look.

**Comment by swaroop-sridhar** [comment]
This issue is likely a manifestation of #21964, and appears to be fixed by: [`1e1f7d3`](/dotnet/coreclr/commit/1e1f7d3a88cb91605a3c9716a5354a64ed707a4f).

I can no longer repro this failure locally.
The lab jobs since Feb 6th have passed this test: 
https://ci.dot.net/job/dotnet_coreclr/job/master/job/jitstress/job/arm64_cross_checked_ubuntu16.04_gcstress0xc_tst/

So, I'll close this issue.
